src/MainWindow.vala

```diff
@@ -77,6 +79,13 @@ public class MainWindow : Hdy.Window {
             return false;
         });
 
+        deck.notify["transition-running"].connect (() => {
+            if (!deck.transition_running) {
+                if (deck.visible_child == welcome_view)
+                    show_welcome_view ();
+            }
```

Comment:
This should not need to be nested

```suggestion
            if (!deck.transition_running && deck.visible_child == welcome_view) {
                show_welcome_view ();
            }
```

Reply:
Yeah. I copied this snippet from a snippet set I have made gathering code from a lot of different projects. Feel free to modify any that you consider inappropriate.